snappy/install.go

```diff
@@ -46,10 +46,46 @@ const (
 	AllowOEM
 )
 
-// Update the installed snappy packages, it returns the updated Parts
+func doUpdate(part Part, flags InstallFlags, meter progress.Meter) error {
+	if _, err := part.Install(meter, flags); err == ErrSideLoaded {
+		logger.Noticef("Skipping sideloaded package: %s", part.Name())
+		return nil
+	} else if err != nil {
+		return err
+	}
+	if err := GarbageCollect(part.Name(), flags, meter); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Update updates the selected name
+func Update(name string, flags InstallFlags, meter progress.Meter) ([]Part, error) {
+	installed, err := NewMetaLocalRepository().Installed()
+	if err != nil {
+		return nil, err
+	}
+	cur := FindSnapsByName(name, installed)
+	if len(cur) != 1 {
+		return nil, ErrNotInstalled
+	}
+
+	// zomg :-(
+	// TODO: query the store for just this package, instead of this
+	updates, err := ListUpdates()
+	upd := FindSnapsByName(QualifiedName(cur[0]), updates)
```

Comment:
Seems strange that we can just call such top level functions without any context and have them work. Eventually we need to move into an API which holds context internally into a value that represents the store.